scraper/scraper.py

```diff
@@ -0,0 +1,212 @@
+import concurrent.futures
+import dotenv
+import itertools
+import json
+import logging
+import os
+import requests
+import sys
+import time
+
+from bs4 import BeautifulSoup
+
+# adjust recursion limit for parsing large files with Beautiful Soup
+sys.setrecursionlimit(10000) # default is 1000
+
+class CoursebookScraper:
+
+    def __init__(self, ptgsessid: str, congfiguration: dict = None, loggingFile: str = 'scraper.log'):
+        # handle object variables
+        self.ptgsessid = ptgsessid
+        self.configuration = congfiguration
+        self.loggingFile = loggingFile
+
+        # initialize logging for scraper
+        self.logger = logging.getLogger('Coursebook-Scraper')
+        logging.basicConfig(
+            format='%(asctime)s - %(levelname)s: %(message)s',
+            datefmt='%Y-%m-%d %H:%M:%S',
+            filename=self.loggingFile, 
+            filemode='w', 
+            level=logging.INFO, 
+        )
+
+        # document object instantiation
+        self.logger.info('Started coursebook scraper')
+
+
+    def getSearchSpecifications(self) -> dict:
```

Comment:
I'll note this now: stick to PEP 8 (the Python style guide) and use that convention for all of the variable/function/class names: https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles